server/notifications.go

```diff
@@ -0,0 +1,95 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package server
+
+import (
+	"context"
+	"sync"
+
+	"github.com/gofrs/uuid"
+)
+
+const (
+	// how many notifications can be in a notification buffer.
+	notificationBufferSize = 16
+)
+
+type change struct {
+	// persistedIndex is a changed persisted index or `nil` if the value has not changed.
+	persistedIndex *uint64
```

Comment:
Why is persisted index a pointer? If there is no update couldn't you just not send a `change{}`?